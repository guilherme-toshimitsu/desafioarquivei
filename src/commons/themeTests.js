import React from "react";
import { ThemeProvider } from "styled-components";
import { render } from "@testing-library/react";
import * as Redux from "react-redux";
import thunk from "redux-thunk";
import theme from "./theme";
import store from "@store";

const renderWithTheme = (children) => {
  return render(<ThemeProvider theme={theme}>{children}</ThemeProvider>);
};

const renderWithThemeAndRedux = (children) => {
  return render(
    <ThemeProvider theme={theme}>
      <Redux.Provider store={store}>{children}</Redux.Provider>
    </ThemeProvider>
  );
};

export { renderWithTheme, renderWithThemeAndRedux };
