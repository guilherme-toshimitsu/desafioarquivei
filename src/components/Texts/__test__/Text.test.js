import React from "react";
import { Text, Title } from "../";
import { renderWithTheme } from "@commons/themeTests";

describe("<Text />", () => {
  it("creates a Text component", () => {
    const wrapper = renderWithTheme(<Text />);
    expect(wrapper.asFragment()).toMatchSnapshot();
  });
});

describe("<Title />", () => {
  it("creates a Title component", () => {
    const wrapper = renderWithTheme(<Title />);
    expect(wrapper.asFragment()).toMatchSnapshot();
  });
});
