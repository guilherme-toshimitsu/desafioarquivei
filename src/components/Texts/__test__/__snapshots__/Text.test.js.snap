// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> creates a Text component 1`] = `
<DocumentFragment>
  <p
    class="Texts__Text-pq9w5c-1 khPFWE"
  />
</DocumentFragment>
`;

exports[`<Title /> creates a Title component 1`] = `
<DocumentFragment>
  <p
    class="Texts__Title-pq9w5c-0 fiQxoW"
  />
</DocumentFragment>
`;
